### PR TITLE
chore: components dedup

### PR DESCRIPTION
**Pull request by Xaroz**
Needs [#4847](https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4847) and  [#4852](https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4852)  from monorepo

- Remove and update components with the new changes from widgets lib
- Use modal from widgets lib
- Use IconButton from widgets lib
- Removed Spinner and replaced with widgets lib

### CODE REVIEW

File: src/components/layout/Modal.tsx [hunk 1]
```diff
@@ -1,7 +1,6 @@
-import { Dialog, Transition } from '@headlessui/react';
+import { Dialog, DialogPanel, DialogTitle, Transition, TransitionChild } from '@headlessui/react';
+import { IconButton, XCircleIcon } from '@hyperlane-xyz/widgets';
 import { Fragment, PropsWithChildren } from 'react';
-import XCircle from '../../images/icons/x-circle.svg';
-import { IconButton } from '../buttons/IconButton';
 
 export function Modal({
```
Comment (jmrossy) [inline src/components/layout/Modal.tsx]: Let's remove this component entirely and use the widgets one as I did here: https://github.com/hyperlane-xyz/hyperlane-explorer/pull/114/files#diff-8d43041520d8da688a2e88756b320ba3a8384e7617a930ae8900009eacef0ecc

**Review by jmrossy — COMMENTED** [review]
Good progress!
Please also de-dupe the Spinner and more of the SVG files under `/src/images/`

File: src/features/wallet/SideBarMenu.tsx [hunk 2]
```diff
@@ -210,31 +208,15 @@ function TransferSummary({
         {STATUSES_WITH_ICON.includes(status) ? (
           <Image src={getIconByTransferStatus(status)} width={25} height={25} alt="" />
         ) : (
-          <SmallSpinner className="-ml-1 mr-3" />
+          <>
```
Comment (jmrossy) [inline src/features/wallet/SideBarMenu.tsx]: Why the fragment?

**Review by jmrossy — COMMENTED** [review]
Excited to remove all this stuff!

File: src/features/wallet/SideBarMenu.tsx (hunk 2 above)
Comment (Xaroz) [reply src/features/wallet/SideBarMenu.tsx]: derp, forgot to remove it after testing something

**Pull request merged**